### PR TITLE
Sync chainable in v4 with lodash master. Fixes #49

### LINKED ISSUE #49: flatMap not detected as part of a lodash chain
this
```
_.chain(a).flatMap('b').map('c').value()
```
is hitting prefer-lodash-method because `flatMap` isn't marked chainable.

But am also confused by the lodash documentation - I raised this..
https://github.com/lodash/lodash/issues/2166
because it looks like everything is chainable.. so am not sure whether everything should be chainable or just to add `flatMap` to the chainable list..

### PR DESCRIPTION
**Pull request by lukeapage**
I just synced with master so its possible it includes some unreleased methods and it definitely includes some methods that aren't added (I raised a bug about that)

### CODE REVIEW

File: src/util/methodDataByVersion.js [hunk 1]
```diff
@@ -507,10 +507,16 @@ module.exports = {
             'dropRight',
             'dropRightWhile',
             'dropWhile',
+            'extend',
```
Comment (lukeapage) [inline src/util/methodDataByVersion.js]: do aliases go in the chain list?

File: src/util/methodDataByVersion.js [hunk 2]
```diff
@@ -507,10 +507,16 @@ module.exports = {
...
+            'extendWith',
             'fill',
             'filter',
+            'flatMap',
+            'flatMapDeep',
```
Comment (lukeapage) [inline src/util/methodDataByVersion.js]: next 2 only in master it seems like

File: src/util/methodDataByVersion.js (hunk 1 above)
Comment (lukeapage) [inline src/util/methodDataByVersion.js]: looks like they don't

**Comment by lukeapage** [comment]
Fixed to include only those methods in 4.6.1 and removed the 2 aliases in the lodash list.

**Comment by ganimomer** [comment]
Merged and released in `v1.5.2`

**Pull request merged**